### PR TITLE
test(cli): migrate dotenv tests to jest from mocha

### PR DESCRIPTION
**Pull request by jasonkuhrt**
Mostly test migration but also some test context utilty work.

### CODE REVIEW

File: src/packages/cli/package.json [hunk 1]
```diff
@@ -123,20 +123,12 @@
...
-    "test:format": "jest format",
-    "test:doctor": "jest doctor",
     "test:studio": "./node_modules/.bin/mocha src/__tests__/studio.test.ts --require ts-mocha src/__tests__/studio.test.ts --timeout 10s",
     "test:sqlite": "./node_modules/.bin/mocha src/__tests__/integrate.sqlite.test.ts --require ts-mocha src/__tests__/integrate.sqlite.test.ts --timeout 10s",
     "test:mariadb": "./node_modules/.bin/mocha src/__tests__/integrate.mariadb.test.ts --require ts-mocha src/__tests__/integrate.mariadb.test.ts --timeout 10s",
     "test:mysql": "./node_modules/.bin/mocha src/__tests__/integrate.mysql.test.ts --require ts-mocha src/__tests__/integrate.mysql.test.ts --timeout 10s",
     "test:postgresql": "./node_modules/.bin/mocha src/__tests__/integrate.postgresql.test.ts --require ts-mocha src/__tests__/integrate.postgresql.test.ts --timeout 10s",
-    "test:commands": "./fixtures/test.sh && pnpm run test:generate && pnpm run test:dotenv && pnpm run test:format && pnpm run test:studio",
+    "test:commands": "./fixtures/test.sh && pnpm run test:generate && jest dotenv- format doctor && pnpm run test:studio",
```
Comment (jasonkuhrt) [inline src/packages/cli/package.json:131]: Nice benefit is faster test run b/c less process spawn

File: src/packages/cli/.gitignore [hunk 2]
```diff
@@ -11,7 +11,6 @@ download-build
 runtime
 examples
 !sqlite.test.db
-!src/__tests__/fixtures/studio-test-project/dev.db
```
Comment (jasonkuhrt) [inline src/packages/cli/.gitignore:14]: This didn't seem to be working so instead I just renamed to `db.sqlite`. Thoughts?

**Comment by jasonkuhrt** [comment]
added generate and studio too now

File: src/packages/cli/.gitignore (hunk 2 above)
Comment (Jolg42) [reply src/packages/cli/.gitignore:14]: all our sqlite db are name `name.db` so maybe don't rename only one

File: src/packages/cli/src/__tests__/__helpers__/context.ts [hunk 3]
```diff
@@ -1,18 +1,42 @@
...
+   *
+   * For this to work the source must be built!
+   */
+  cli: (...input: string[]) => ExecaChildProcess<string>
+}
+
+/**
+ * Create test context to use in tests. Provides the following:
+ *
+ * - A temporary diectory
```
Comment (Jolg42) [inline src/packages/cli/src/__tests__/__helpers__/context.ts:33]:
```suggestion
 * - A temporary directory
```

File: src/packages/cli/src/__tests__/__helpers__/context.ts [hunk 4]
```diff
@@ -32,6 +47,48 @@ export const Context = {
...
+    assemble(): Context {
+      return ctx
+    },
+  }
+}
+
+/**
+ * Test context contributor. Mocks console.error with a Jest spy before each test.
+ */
+export const consoleContext: ContextContributor<
```
[the remark this answers is not in the document]
Comment (Jolg42) [reply src/packages/cli/src/__tests__/__helpers__/context.ts:98]: I will actually use a mock version for console.log in a test today, it's 💯

[the remark this answers is not in the document]
Comment (Jolg42) [reply src/packages/cli/src/__tests__/__helpers__/context.ts:98]: So how would I be able to use this for console.log?

[the remark this answers is not in the document]
Comment (jasonkuhrt) [reply src/packages/cli/src/__tests__/__helpers__/context.ts:98]: I think the easiest thing to do would be enhance the context contributor to mock both `console.error` and `console.log`.

I didn't do this yet but I think context contributors should be wrapped in a factory function allowing the context contributor itself to be configured.

Even context contributors that don't need config should be wrapped in factors so it is consistent.

Then we could do something like this:

```ts
// imagine: default consoleContext options are log:true, error:true
Context.new().add(consoleContext({ log: false }))
```

File: src/packages/cli/.gitignore (hunk 2 above)
Comment (jasonkuhrt) [reply src/packages/cli/.gitignore:14]: Cool I'll stick to that convention!

File: src/packages/cli/src/__tests__/__helpers__/context.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (Jolg42) [reply src/packages/cli/src/__tests__/__helpers__/context.ts:98]: Sounds good to me, possible to do it soon?

[the remark this answers is not in the document]
Comment (jasonkuhrt) [reply src/packages/cli/src/__tests__/__helpers__/context.ts]: @Jolg42 took a stab at it (no explicit options yet) in this PR.

File: src/packages/cli/src/__tests__/generate.test.ts [hunk 5]
```diff
@@ -14,7 +14,7 @@ it('should work with a custom output dir', async () => {
   const { main } = await import(ctx.fs.path('main.ts'))
   expect(cleanSnapshot(data.stdout)).toMatchSnapshot()
   await expect(main()).resolves.toMatchSnapshot()
-})
+}, 10000)
```
Comment (Jolg42) [inline src/packages/cli/src/__tests__/generate.test.ts:17]: generate should be way faster than that less than a second on most all machines so maybe changing it like this:
```suggestion
}, 1000)
```

File: src/packages/cli/src/bin.ts [hunk 6]
```diff
@@ -1,25 +1,50 @@
...
+import path from 'path'
+import { CLI } from './CLI'
+import { Dev } from './Dev'
+import { Doctor } from './Doctor'
+import { Format } from './Format'
+import { Generate } from './Generate'
+import { Studio } from './Studio'
+import { Telemetry } from './Telemetry'
+import { Validate } from './Validate'
+import { Version } from './Version'
```
[the remark this answers is not in the document]
Comment (Jolg42) [reply src/packages/cli/src/bin.ts:42]: How do you activate this?

**Review by Jolg42 — APPROVED** [review]
Awesome work on these tests 💚 

Can be merged after tests are passing!

File: src/packages/cli/src/__tests__/generate.test.ts (hunk 5 above)
Comment (jasonkuhrt) [reply src/packages/cli/src/__tests__/generate.test.ts]: image

hmm

Comment (Jolg42) [reply src/packages/cli/src/__tests__/generate.test.ts]: Oh sorry then....

File: src/packages/cli/src/bin.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (jasonkuhrt) [reply src/packages/cli/src/bin.ts]: image

File: src/packages/cli/src/__tests__/generate.test.ts [hunk 7]
```diff
@@ -1,67 +1,24 @@
...
-        },
-      )
-    } catch (e) {
-      assert.equal(e.exitCode, 1)
-    }
-  })
+  const { main } = await import(ctx.fs.path('main.ts'))
+  expect(cleanSnapshot(data.stdout)).toMatchSnapshot()
+  await expect(main()).resolves.toMatchSnapshot()
+}, 1000)
```
Comment (Jolg42) [inline src/packages/cli/src/__tests__/generate.test.ts:17]:
```suggestion
}, 10000)
```

File: src/packages/cli/src/bin.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (jasonkuhrt) [reply src/packages/cli/src/bin.ts]: https://code.visualstudio.com/docs/languages/typescript#_organize-imports

**Pull request merged**